<a href="https://colab.research.google.com/github/Aayushi-Sharma24/EDA-Hotel-Booking-Analysis/blob/main/Hotel_Booking_Analysis_Capstone_Project_Final.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Have you ever wondered when the best time of year to book a hotel room is? Or the optimal length of stay in order to get the best daily rate? What if you wanted to predict whether or not a hotel was likely to receive a disproportionately high number of special requests? This hotel booking dataset can help you explore those questions!

## <b>This data set contains booking information for a city hotel and a resort hotel, and includes information such as when the booking was made, length of stay, the number of adults, children, and/or babies, and the number of available parking spaces, among other things. All personally identifying information has been removed from the data. </b>

## <b> Explore and analyze the data to discover important factors that govern the bookings. </b>

#Introduction
###Hotel industry is very volatile industry and the bookings depend on variety of factors such as type of hotels, seasonality, days of week and many more. This make analyzing the patterns available in the past data more important to help hotels plan better.

#Let's have a look at Column Information
**Hotel**: types of hotel

**is_canceled**: booking canceled or not

**lead_time**: No. of days before actual arrival in the hotel

**arrival_date_year**: year of booking

**arrival_date_month**: month of booking

**arrival_date_week_number**: week number of the year of booking

**arrival_date_day_of_month**: arrival month date

**stays_in_weekend_nights**: no. of weekends guests stayed

**stays_in_week_nights**: no. of weekdays guests stayed

**adults**: No. of Adults

**children**: No. of children

**babies**: No. of babaies

**meal**: Type of meal booked. 

*   Undefined/SC – no meal package; 
*   BB – Bed & Breakfast;
*   HB – Half board (breakfast and one other meal);
*   FB – Full board (breakfast, lunch and dinner)

**country**: country from which guests belong

**market_segment**: group of people who share one or more common characteristics, lumped together for marketing purposes

*   TA: Travel agents
*   TO: Tour operators

**distribution_channel**: chain of businesses

*   TA: Travel agents
*   TO: Tour operators

**is_repeated_guest**:  booking name was from repeated guest or not

**previous_cancellations**:  No. of previous bookings that were cancelled by the customer prior to the current booking

**previous_bookings_not_canceled**:  No. of previous bookings not cancelled by the customer prior to the current booking

**reserved_room_type**:  Code of room type reserved

**assigned_room_type**:  Code for the type of room assigned to the booking

**booking_changes**:  No. of changes made to the booking

**deposit_type**: 
*   Deposit – no deposit was made;
*   Non Refund 
*   Refundable

**agent**: ID of the travel agency that made the booking

**company**: ID of the company that made the booking

**day_in_waiting_list**: No. of days the booking was in the waiting list before it was confirmed to the customer

**customer_type**: type of customer

**adr (average daily rate)**

**required_car_parking_spaces**: No. of car parking spaces required by the customer

**total_of_special_requests**: No. of special requests made by the customer 

**reservation_status**: 
*   Canceled – booking was canceled by the customer;
*   Check-Out – customer has checked in but already departed;
*   No-Show – customer did not check-in 

**reservation_status_date**: Date at which the last status was set

#Objective
### To create an indepth analysis to figure out the standard patterns of booking based on various factors

# Data Preparation and Cleaning

### Let's import the required libraries that we will be using for this project

In [ ]:
# Importing required packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

## Importing dataset into notebook

In [ ]:
# Mount drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


###Let's Download the dataset to use it as a pandas dataframe

In [ ]:
#Loading csv file to dataframe
path = "/content/drive/MyDrive/Colab Notebooks/project EDA/Hotel Bookings.csv"
df = pd.read_csv(path)

In [ ]:
df.head()

hotel  is_canceled  ...  reservation_status  reservation_status_date
0  Resort Hotel            0  ...           Check-Out               2015-07-01
1  Resort Hotel            0  ...           Check-Out               2015-07-01
2  Resort Hotel            0  ...           Check-Out               2015-07-02
3  Resort Hotel            0  ...           Check-Out               2015-07-02
4  Resort Hotel            0  ...           Check-Out               2015-07-03

[5 rows x 32 columns]

##Let's Explore the DataFrame
###Looking into shape of data to find out number of rows and columns

In [ ]:
# checks the no. of rows and columns
df.shape

(119390, 32)

 So there are 119390 rows and 32 columns in our DataFrame

### Let's have  look at all the column names present in our DataFrame

In [ ]:
# Check column names
df.columns

Index(['hotel', 'is_canceled', 'lead_time', 'arrival_date_year',
       'arrival_date_month', 'arrival_date_week_number',
       'arrival_date_day_of_month', 'stays_in_weekend_nights',
       'stays_in_week_nights', 'adults', 'children', 'babies', 'meal',
       'country', 'market_segment', 'distribution_channel',
       'is_repeated_guest', 'previous_cancellations',
       'previous_bookings_not_canceled', 'reserved_room_type',
       'assigned_room_type', 'booking_changes', 'deposit_type',
       'days_in_waiting_list', 'customer_type', 'adr',
       'required_car_parking_spaces', 'total_of_special_requests',
       'reservation_status', 'reservation_status_date'],
      dtype='object')

 ### Looking into the datatype of each column

In [ ]:
# checks the datatypes of each column
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 118898 entries, 0 to 119389
Data columns (total 30 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   hotel                           118898 non-null  object 
 1   is_canceled                     118898 non-null  int64  
 2   lead_time                       118898 non-null  int64  
 3   arrival_date_year               118898 non-null  int64  
 4   arrival_date_month              118898 non-null  object 
 5   arrival_date_week_number        118898 non-null  int64  
 6   arrival_date_day_of_month       118898 non-null  int64  
 7   stays_in_weekend_nights         118898 non-null  int64  
 8   stays_in_week_nights            118898 non-null  int64  
 9   adults                          118898 non-null  int64  
 10  children                        118898 non-null  float64
 11  babies                          118898 non-null  int64  
 12  meal            